File: backend/app/schemas/multimodal.py
from pydantic import BaseModel
from typing import Optional

class AnalyzeRequest(BaseModel):
    image: str  # base64 encoded image
    prompt: Optional[str] = "Describe this data pipeline diagram in detail."

class AnalyzeResponse(BaseModel):
    success: bool
    analysis: Optional[str] = None
    model: Optional[str] = None
    tokens: Optional[int] = None
    error: Optional[str] = None

class MultimodalStatus(BaseModel):
    available: bool
    mode: str = "local"  # "local", "remote", "disabled"
    model: Optional[str] = None
    loaded: bool = False
    remote_health: Optional[str] = None

File: frontend/src/api/multimodal.ts
import api from './index';

export interface AnalyzeRequest {
  image: string;  // Base64 data URL
  prompt?: string;
  temperature?: number;
  max_tokens?: number;
}

export interface AnalyzeResponse {
  success: boolean;
  analysis?: string;
  error?: string;
  model?: string;
  prompt?: string;
  tokens?: number;
}

export const multimodalApi = {
  analyze: async (request: AnalyzeRequest): Promise<AnalyzeResponse> => {
    const response = await api.post('/api/v1/multimodal/analyze', request);
    return response.data;
  },

  uploadAndAnalyze: async (file: File, prompt?: string): Promise<AnalyzeResponse> => {
    const formData = new FormData();
    formData.append('file', file);
    if (prompt) formData.append('prompt', prompt);

    const response = await api.post('/api/v1/multimodal/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    });
    return response.data;
  },

  getStatus: async (): Promise<{ available: boolean; model?: string }> => {
    const response = await api.get('/api/v1/multimodal/status');
    return response.data;
  },
};
